### PR TITLE
birthday

### PR DESCRIPTION
**Pull request by azylman**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const async        = require('async');
+const EventEmitter = require('events');
+const crypto       = require('./lib/crypto.js');
+
+module.exports = class SessionEncrypt extends EventEmitter {
+  constructor(key, store) {
+    super();
+    if (key.length != 32) {
+      throw new Error("Invalid key length: must be 32 bytes");
```
Comment (rgarcia) [inline index.js:11]: string length != byte length, so this is confusing...

Comment (rgarcia) [inline index.js:11]: http://stackoverflow.com/questions/9864662/how-to-get-the-string-length-in-bytes-in-nodejs/9864762#9864762

File: lib/crypto.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if (version !== supportedVersion) {
+      throw new Error("Invalid version number " + version);
+    }
+    let cipher = crypto.createDecipheriv("aes-256-gcm", key, new Buffer(iv, "hex"));
+    cipher.setAuthTag(new Buffer(tag, "hex"));
+    let plainText = cipher.update(cipherText, "hex", "utf8");
+    plainText += cipher.final("utf8");
+    return plainText;
+  }
+}
```
Comment (rgarcia) [inline lib/crypto.js:25]: all the `let`s can be `const`

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const assert                = require('assert');
+const sinon                 = require('sinon');
+const ConnectSessionEncrypt = require('../index.js');
+
+const KEY = "https://i.imgur.com/YSho6WQ.jpeg";
```
Comment (rgarcia) [inline test/index.js:7]: epic test data

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  });
+
+  describe("encryption", () => {
+    it("can round-robin", () => {
+      let csc = new ConnectSessionEncrypt(KEY, {});
+      assert.equal(csc.decrypt(csc.encrypt("data")), "data");
+      assert.notEqual(csc.encrypt("data"), "data");
+    });
+  });
+});
```
Comment (rgarcia) [inline test/index.js:102]: i think you can s/let/const/ in this file

**Comment by rgarcia** [comment]
some nits otherwise lgtm

File: index.js (hunk 1 above)
Comment (azylman) [inline index.js:11]: as long as they're only using ASCII... ;)

**Pull request merged**